tests/vectors_kem.c

```diff
@@ -58,13 +77,75 @@ static void hexStringToByteArray(const char *hexString, uint8_t *byteArray) {
 	}
 }
 
-/* ML_KEM-specific functions */
-static inline bool is_ml_kem(const char *method_name) {
-	return (0 == strcmp(method_name, OQS_KEM_alg_ml_kem_512))
-	       || (0 == strcmp(method_name, OQS_KEM_alg_ml_kem_768))
-	       || (0 == strcmp(method_name, OQS_KEM_alg_ml_kem_1024));
+#ifdef OQS_ENABLE_KEM_ML_KEM
+static inline bool sanityCheckSK(const uint8_t *sk, const char *method_name) {
+	/* sanity checks */
+	if ((NULL == sk) || (NULL == method_name) || (false == is_ml_kem(method_name))) {
+		fprintf(stderr, "[vectors_kem] %s ERROR: inputs NULL or invalid method !\n", method_name);
+		return false;
+	}
+	/* buffer to hold public key hash */
+	uint8_t pkdig[SHA256_OP_LEN] = {0};
+	/* fetch the value of k according to the ML-KEM algorithm as per FIPS-203
+	K = 2 for ML-KEM-512, K = 3 for ML-KEM-768 & K = 4 for ML-KEM-1024 */
+	uint8_t K = (0 == strcmp(method_name, OQS_KEM_alg_ml_kem_512)) ? 2 : (0 == strcmp(method_name, OQS_KEM_alg_ml_kem_768)) ? 3 : 4;
```

Comment:
I love the ternary operator, but this logic would be much easier to parse if it used `if/else`.

Reply:
moved it to a separate function now.